### ISSUE TITLE
Dockerfile.native runtime base image for mostly static native executables

### ISSUE THREAD

**Issue by romain-grecourt** [helidon-io/helidon#6283]
Since we switched to the official `native-maven-plugin` (done as part of https://github.com/helidon-io/helidon/pull/5308) the native executables built are now "mostly static" thus are linked with `glibc`.

We still have `-Dnative.image.buildStatic` in our `Dockerfile.native`, however this option is only processed by `helidon:native-image`.

See:
- https://www.graalvm.org/22.1/reference-manual/native-image/StaticImages
- https://www.graalvm.org/22.2/reference-manual/native-image/guides/containerise-native-executable-and-run-in-docker-container

We need to decide if we want to keep build "static" or if we want to switch to "mostly static".

---

If we decide to revert back to "static" we'd have to configure the `native-maven-plugin` like this:

```xml
<plugin>
    <groupId>org.graalvm.buildtools</groupId>
    <artifactId>native-maven-plugin</artifactId>
    <configuration>
        <buildArgs>--static</buildArgs>
    </configuration>
</plugin>
```

---

If we were to switch to "mostly static",  then we would have to change the runtime base image.

Ideally we'd be using `FROM gcr.io/distroless/base` ; however this fails:

```
 error while loading shared libraries: libz.so.1: cannot open shared object file: No such file or directory
Error: exit status 127
```

Another option would be to use slim images:
- `oraclelinux:9-slim`
- `ghcr.io/oracle/oraclelinux:9-slim`

I deliberately listed ol9 images to match the build stage base image `ghcr.io/graalvm/graalvm-ce:ol9-java17-22.3.0`

**Comment by romain-grecourt** [comment]
Note that I don't think we can use distroless images as they are debian based and our build base image is based on oraclelinux.

I've tried to update the runtime stage to include the required libraries like this:

```Dockerfile
FROM gcr.io/distroless/base:debug
WORKDIR /helidon

# Copy the binary built in the 1st stage
COPY --from=build /lib64/libz* /lib/x86_64-linux-gnu/
COPY --from=build /lib64/libstdc* /lib/x86_64-linux-gnu/
COPY --from=build /lib64/libgcc* /lib/x86_64-linux-gnu/
COPY --from=build /helidon/target/mp-native .

ENTRYPOINT ["./mp-native"]

EXPOSE 8080
```

It failed with:

```bash
./mp-native: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by ./mp-native)
./mp-native: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by ./mp-native)
./mp-native: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /lib/x86_64-linux-gnu/libstdc++.so.6)
./mp-native: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /lib/x86_64-linux-gnu/libstdc++.so.6)
./mp-native: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /lib/x86_64-linux-gnu/libstdc++.so.6)
./mp-native: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /lib/x86_64-linux-gnu/libgcc_s.so.1)
```

So... the bundled `glibc` version is not enough.

distroless provides `2.30`:

```bash
nerdctl run --rm -it --entrypoint sh gcr.io/distroless/base:debug -c "strings /lib/x86_64-linux-gnu/libc.so.6 | grep -E '^GLIBC_' | sed s/GLIBC_//g |  sort -V | tail -3"
2.28
2.29
2.30
```

Our build image provides `2.34`:

```bash
nerdctl run --rm -it --entrypoint sh ghcr.io/graalvm/graalvm-ce:ol9-java17-22.3.0 -c "strings /lib64/libc.so.6 | grep -E '^GLIBC_' | grep -v PRIVATE |  sed s/GLIBC_//g | sort -V | tail -3"
2.33
2.33
2.34
```

Trying to copy libc from the build image leads to other errors...

---

Bottom line, we need update our build image to something compatible with `distroless/base` in order to use `distroless/base`.

**Comment by romain-grecourt** [comment]
see also https://github.com/helidon-io/helidon/issues/6260
